nemo/collections/asr/parts/utils/streaming_utils.py

```diff
@@ -1608,28 +1609,44 @@ def get_all_audios(self):
 class FrameBatchMultiTaskAED(FrameBatchASR):
     def __init__(self, asr_model, frame_len=4, total_buffer=4, batch_size=4):
         super().__init__(asr_model, frame_len, total_buffer, batch_size, pad_to_buffer_len=False)
+        self.window_stride = asr_model._cfg.preprocessor.window_stride
+        self.subsampling_factor = asr_model._cfg.encoder.subsampling_factor
+        self.chunk_offsets = [0,]     # chunk offsets in terms of num frames before subsampling
 
+    def reset(self):
+        super().reset()
+        self.chunk_offsets = [0,]
+    
     def get_input_tokens(self, sample: dict):
         if self.asr_model.prompt_format == "canary":
-            missing_keys = [k for k in ("source_lang", "target_lang", "taskname", "pnc") if k not in sample]
-            if missing_keys:
-                raise RuntimeError(
-                    f"We found sample that is missing the following keys: {missing_keys}"
-                    f"Please ensure that every utterance in the input manifests contains these keys. Sample: {sample}"
-                )
-            tokens = self.asr_model.prompt.encode_dialog(
-                turns=[
-                    {
-                        "role": "user",
-                        "slots": {
-                            **sample,
-                            self.asr_model.prompt.PROMPT_LANGUAGE_SLOT: "spl_tokens",
-                        },
-                    }
-                ]
-            )["context_ids"]
+            expected_slots = {"source_lang", "target_lang", "taskname", "pnc"}
+        elif self.asr_model.prompt_format == "canary2":
+            expected_slots = {"source_lang", "target_lang"}
         else:
             raise ValueError(f"Unknown prompt format: {self.asr_model.prompt_format}")
+        
+        missing_keys = [k for k in expected_slots if k not in sample]
+        if missing_keys:
+            raise RuntimeError(
+                f"We found sample that is missing the following keys: {missing_keys}"
+                f"Please ensure that every utterance in the input manifests contains these keys. Sample: {sample}"
+            )
+        
+        # fill optional slots
+        for k, v in self.asr_model.prompt.get_default_slot_values().items():
```

Comment:
Please test it with canary-1b - it will likely fail, because PromptFormatter has no `get_default_slot_values` method. 

Let's try to implement this without changing PromptFormatter's API. It's a deliberate choice not to assign default slot values to prompt formatter class; instead each trained model should define its own default values (they can be different with the same prompt format). If you need to hardcode some here to get this released quickly, I am OK with that.